### PR TITLE
chore: bump dpy version

### LINKED ISSUE #315: Bump discord-py version
Recently, discord-py 2.6.X came out. Supporting the components v2 system as well as other improvements. Currently, "blocked" by a typing error https://github.com/Rapptz/discord.py/issues/10280.

You could always just ignore this error (present in the move_messages file) and upgrade anyway. Started bump but stopped after typing error, so unknown what other work is actually needed (if any). 

There are more changes coming to modals in about a month. Incremental updates are still the easier solution; so I suggest upgrading now instead of waiting for all changes to come in. 

Will keep you updated on the state of my bug report. And can create a PR for bumping discord-py lib versions.

### PR DESCRIPTION
**Pull request by CoderJoshDK**
Closes #315

See the changelogs for a full idea of what new features exist / bug fixes implemented https://discordpy.readthedocs.io/en/stable/whats_new.html#v2-6-0

Tested that things still work as expected. However, I do not have a full test suite checklist to ensure that actually all edge cases are still working as expected.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ authors = [
   { name = "trag1c", email = "dev@jakubr.me" },
 ]
 dependencies = [
-  "discord-py~=2.5.0",
+  "discord-py~=2.6.0",
```
Comment (00-kat) [inline pyproject.toml:13]: This should probably be:

```suggestion
  "discord-py>=2.6.2,<2.7",
```

Due to https://github.com/Rapptz/discord.py/issues/10280.

**Review by trag1c — APPROVED** [review]
